kv/bufpool.go

```diff
@@ -0,0 +1,44 @@
+package kv
+
+import (
+	"sync"
+
+	"github.com/ngaut/log"
+)
+
+// A cache holds a set of reusable objects.
+// The slice is a stack (LIFO).
+// If more are needed, the cache creates them by calling new.
+type cache struct {
+	mu    sync.Mutex
+	name  string
+	saved []MemBuffer
+	new   func() MemBuffer
+}
+
+func (c *cache) put(x MemBuffer) {
+	c.mu.Lock()
+	if len(c.saved) < cap(c.saved) {
+		c.saved = append(c.saved, x)
+	} else {
+		log.Warning(c.name, "is full, you may need to increase pool size")
```

Comment:
I think `log.Warnf("%s is full, you may need to increase pool size", c.name)` is better.